context/trace_test.go

```diff
@@ -68,18 +68,21 @@ type valueTestCase struct {
 	notnilorempty bool // just check not empty/not nil
 }
 
-func checkContextForValues(ctx context.Context, t *testing.T, values []valueTestCase) {
-	for _, testcase := range values {
-		v := ctx.Value(testcase.key)
-		if testcase.notnilorempty {
-			if v == nil || v == "" {
-				t.Fatalf("value was nil or empty for %q: %#v", testcase.key, v)
+func checkContextForValues(ctx context.Context, t *testing.T, tests []valueTestCase) {
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.key, func(t *testing.T) {
```

Comment:
I am not sure I like the idea of calling a subtest from inside a private `func`. I like to be explicit about these thiings and call them inside the `Test` funcs. Also, we could parallelise this too 🤔

Reply:
Yeah; looks like this function was written to make the test more DRY, but effectively this "helper" was running subtests. Changing it to be `TestXX` wouldn't work because it doesn't have the right signature, so I just pushed a commit that inlines what it does (at cost of duplicating a couple of lines of code)